### PR TITLE
synaptics-rmi: add support to write signature

### PR DESCRIPTION
**Pull request by vhuag**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [x] Feature
- [ ] Documentation
When the firmware partition contains signature, firmware would require an additional step to write signature.

### CODE REVIEW

File: plugins/synaptics-rmi/fu-synaptics-rmi-v7-device.c [hunk 1]
```diff
@@ -331,6 +416,14 @@ fu_synaptics_rmi_v7_device_write_partition(FuSynapticsRmiDevice *self,
 			return FALSE;
 		fu_progress_step_done(progress);
 	}
+	if (!fu_synaptics_rmi_v7_device_write_partition_signature(self,
+								  firmware,
+								  id,
+								  partition_id,
+								  fu_progress_get_child(progress),
```
Comment (hughsie) [inline plugins/synaptics-rmi/fu-synaptics-rmi-v7-device.c:423]: I don't think this is going to work, I think you need to split up the phases with a new `fu_progres_set_id()` block and some new steps. If you push this branch to the origin remote I can push a fixup if that helps.

Comment (vhuag) [reply plugins/synaptics-rmi/fu-synaptics-rmi-v7-device.c:423]: Hi do I need permission to push to origin?

Comment (hughsie) [reply plugins/synaptics-rmi/fu-synaptics-rmi-v7-device.c:423]: You're in the commit list, so just push something like wip/vhuag/linux-add-signature-20230109

Comment (vhuag) [reply plugins/synaptics-rmi/fu-synaptics-rmi-v7-device.c]: Hi I can't push due to permission so I remove the progress tracking inside the signature function, normally the chunks->len is 1.

**Review by hughsie — APPROVED** [review]

**Pull request merged**